### DISCUSSION TITLE
custom board configuration 32MZ2048EFH100.

### DISCUSSION THREAD

**Discussion by JuanDValenciano** [chipKIT32/chipKIT-core#469] [Q&A]
Hello group

we design a board based on the PIC32MZ2048EFH100 and we would like to use this framework. in our board we have a 12Mhz crystal and the one that supports this framework is the board [FLIP&CLICK PIC32MZ](https://www.mikroe.com/flipclick-pic32mz) based on the same MCU that uses a 24Mhz crystal. i was looking for the registers to make the changes but i can't find them. could you guide me to make the change?

**Comment by majenkotech** [comment]
These registers are part of the chip configuration and are normally set by the bootloader (since they form part of the boot flash). Chapter 34 of the datasheet details these configuration registers, and it would be the FPLLIDIV and UPLLFSEL of DEVCFG2 that you would want to tweak.

**Reply by JuanDValenciano** [reply]
Thank you for your reply @majenkotech  

I was thinking about a version without the bootloader for the moment. 

These MCUs have a separate flash space of 2048Kbyte that could be used to mount the bootloader. The devs of [FLIP&CLICK PIC32MZ](https://www.mikroe.com/flipclick-pic32mz) reserve 16Kbytes of the 2048Kbytes in the main flash and I would like to have all the flash available.

![image](https://user-images.githubusercontent.com/15575065/139111712-bad75963-37f8-4c12-a9f0-77a9bde12d12.png)

Are the bootloaders of the boards supported by [chipKIT-core](https://github.com/chipKIT32/chipKIT-core) open-source? where can I download them to have a look?

**Comment by majenkotech** [comment]
Then you need to feed the equivalent hex value for the registers into the linker script. I use mplab-x to generate the configuration and work out what the values should be.

**Reply by JuanDValenciano** [reply]
I recently found the bootloader repo:
https://github.com/chipKIT32/PIC32-avrdude-bootloader

I just did the first tests configuring the pin/out with Harmony V3, the linker file that he generates. could I use it ? technically I copied and pasted the linker file but it doesn't work. should I use Harmony or a standalone version ?

**Reply by majenkotech** [reply]
The bootloader system for chipKIT is unique to chipKIT. It gets more complicated if you don't have the bootloader (at least for MZ chips) as the default linker scripts wipe out the config settings. You need to have a "chip" linker script (check the chipKIT variant files for the F&C in chipKIT) along with a "common" linker script that doesn't wipe the config settings. TBH that side of things is a bit of a mess at the moment. 

Then of course there's the two different bootloaders: the one you pointed to is UART only, but is the "default" one for chipKIT. The alternative is my HID based bootloader which is compiled as a chipKIT sketch and requires a special linker script / board configuration to compile it.
